### PR TITLE
Convert misk-crypto-testing to be a test fixture

### PR DESCRIPTION
**Pull request by tso**

### CODE REVIEW

**Review by frojasg — APPROVED** [review]
Thanks for this change!

File: misk-crypto/src/test/kotlin/misk.crypto/CiphertextFormatTest.kt [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+import misk.testing.MiskTest
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.jupiter.api.Test
+import java.io.ByteArrayOutputStream
+import java.nio.ByteBuffer
+
+@MiskTest
+class CiphertextFormatTest {
```
Comment (swankjesse) [inline misk-crypto/src/test/kotlin/misk.crypto/CiphertextFormatTest.kt:11]: Not net new code right?

File: misk-crypto/src/testFixtures/kotlin/misk/crypto/testing/CryptoTestModule.kt [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+/**
+ * This module should be used for testing purposes only.
+ * It generates random keys for each key name specified in the configuration
+ * and uses [FakeKmsClient] instead of a real KMS service.
+ *
+ * This module **will** read the exact same configuration files as the real application,
+ * but **will not** use the key material specified in the configuration.
+ * Instead, it'll generate a random keyset handle for each named key.
+ */
+class CryptoTestModule(
```
Comment (swankjesse) [inline misk-crypto/src/testFixtures/kotlin/misk/crypto/testing/CryptoTestModule.kt:53]: Love all of this thing

File: misk-crypto/src/testFixtures/kotlin/misk/crypto/testing/CryptoTestModule.kt [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  private val config: CryptoConfig? = null
+) : KAbstractModule() {
+  override fun configure() {
+    bind<KmsClient>().toInstance(FakeKmsClient())
+
+    AeadConfig.register()
```
Comment (swankjesse) [inline misk-crypto/src/testFixtures/kotlin/misk/crypto/testing/CryptoTestModule.kt:59]: Off-topic: side-effects during `configure()` are bad, but that has nothing to do with this PR.

(And I can’t suggest a better alternative; this is the JDK’s fault for having global singletons for crypto things.)

**Review by swankjesse — APPROVED** [review]

File: misk-crypto/src/test/kotlin/misk.crypto/CiphertextFormatTest.kt (hunk 1 above)
Comment (tso) [reply misk-crypto/src/test/kotlin/misk.crypto/CiphertextFormatTest.kt:11]: yeah just copy pasted but i have to leave the old code while i migrate things with topographer

**Pull request merged**